### PR TITLE
Take an `IntoIterator` of `TextArea` in `prepare`

### PR DESCRIPTION
**Pull request by hecrj**
Not too sure about the API changes on this one, but it'd be great to have a way to provide the list of `TextArea` without allocating.

### CODE REVIEW

File: src/text_render.rs [hunk 1]
```diff
@@ -69,7 +69,7 @@ impl TextRenderer {
         queue: &Queue,
         atlas: &mut TextAtlas,
         screen_resolution: Resolution,
-        text_areas: &[TextArea<'a, 'b>],
+        text_areas: impl Iterator<Item = TextArea<'a, 'b>> + Clone,
```
Comment (grovesNL) [inline src/text_render.rs:72]: With the changes from #33 merged, could we remove the `Clone` requirement?

File: src/lib.rs [hunk 2]
```diff
@@ -90,6 +90,7 @@ impl Default for TextBounds {
 }
 
 /// A text area containing text to be rendered along with its overflow behavior.
+#[derive(Clone, Copy)]
```
Comment (grovesNL) [inline src/lib.rs:93]: `Clone` seems ok but could we leave off `Copy` for now? It seems likely that this API could change with future changes to cosmic-text's attributes, so we might not be able to keep it as `Copy` later

File: src/text_render.rs (hunk 1 above)
Comment (hecrj) [reply src/text_render.rs]: You are right! It seems we can merge both loops and iterate only once. Done that in 2f7239b22827fca4ee03666094972ed8d6f9ac7d.

**Comment by hecrj** [comment]
I have rebased on top of `main`. The diff is a bit unwieldy, but it's mostly a bunch of additional nesting.

Are there any other concerns here?

**Review by grovesNL — APPROVED** [review]
Thank you!

**Pull request merged**